### ISSUE TITLE
dotnet installation

### ISSUE THREAD

**Issue by frostworx** [sonic2kk/steamtinkerlaunch#310]
Short problem description in the [DotNet wiki](https://github.com/frostworx/steamtinkerlaunch/wiki/DotNet)
~~While [rewriting](https://github.com/frostworx/steamtinkerlaunch/issues/306) several Vortex functions
a generic **working** installDotNet48 function was added to stl _(not released yet)_~~

A clean _(as possible)_ solution for installing dotnet48 using proton is required.

This could be added at least to all supported 3rd party programs, which use an own [Compatdata](https://github.com/frostworx/steamtinkerlaunch/wiki/Compatdata)/[Wineprefix](https://github.com/frostworx/steamtinkerlaunch/wiki/Wineprefix)
like
- [Vortex](https://github.com/frostworx/steamtinkerlaunch/wiki/Vortex)
- [Mod-Organizer 2](https://github.com/frostworx/steamtinkerlaunch/wiki/Mod-Organizer-2)
- ~~[Depressurizer](https://github.com/frostworx/steamtinkerlaunch/wiki/Depressurizer)~~
**Nope, I do not even use this, chances are higher that it will be removed again**

~~but might be also added as a generic  install option for every [Compatdata](https://github.com/frostworx/steamtinkerlaunch/wiki/Compatdata)/[Wineprefix](https://github.com/frostworx/steamtinkerlaunch/wiki/Wineprefix) of any game.~~

~~Running custom user-installed programs depending on [DotNet](https://github.com/frostworx/steamtinkerlaunch/wiki/DotNet) should be pretty simple then and also allow f.e. to use the [UniversalUE4Unlocker](https://github.com/frostworx/steamtinkerlaunch/wiki/UniversalUE4Unlocker) gui or maybe also use [Vortex](https://github.com/frostworx/steamtinkerlaunch/wiki/Vortex) and [Mod-Organizer 2](https://github.com/frostworx/steamtinkerlaunch/wiki/Mod-Organizer-2) directly in the game pfx without any restrictions.~~

**Nope, as this requires a specific proton/wine version - won't risk breaking game prefixes by mixing proton versions**

**Comment by frostworx** [comment]
Vortex works fine with the installDotNet function, but MO2 doesn't. Looks like the dotnet installation is not 100% clean _(has NETFXRepair.exe)_.

**Comment by frostworx** [comment]
Nope, Vortex works 99%, but it can't extract downloaded mods. https://github.com/frostworx/steamtinkerlaunch/issues/306.
Very likely this is a dotnet issue as well...

**Comment by frostworx** [comment]
Vanilla `wine-6.16` seems to work fine with `winetricks dotnet48`, so the custom `installDotNet` will be removed in favor of switching automatically to a compatible vanilla wine version _(when using a custom compatdata, so at least for [Vortex](https://github.com/frostworx/steamtinkerlaunch/issues/306) and [MO2](https://github.com/frostworx/steamtinkerlaunch/issues/304))_

**Comment by frostworx** [comment]
Just updated [DotNet](https://github.com/frostworx/steamtinkerlaunch/wiki/DotNet)
TL;DR:
The only valid options I currently see 
- wait for official proton wine version >=6.6
- roll a custom proton with an up2date vanilla wine

**Comment by frostworx** [comment]
will use self rolled
https://github.com/frostworx/steamtinkerlaunch-tweaks/releases/tag/Proton-6.17-STL-1
at least until this works with official proton

**Comment by frostworx** [comment]
With [Vortex](https://github.com/frostworx/steamtinkerlaunch/issues/306) and [MO2](https://github.com/frostworx/steamtinkerlaunch/issues/304) seemingly working fine with this solution, I'll close this issue.

The [Vortex](https://github.com/frostworx/steamtinkerlaunch/issues/306) rewrite is pretty much feature complete.
Maybe some more changes to the Mod Organizer 2 (https://github.com/frostworx/steamtinkerlaunch/issues/304)
implementation _(they share many functions now :))_
and then I'll probably push the monster commit somewhere in the 1st third of October.
Any fixable reported issue will make it into the end of October major release _(when I find the time)_
